guide/api-design-guidelines.md

```diff
@@ -0,0 +1,181 @@
+# Radius API Guidelines
+
+<!-- markdownlint-disable MD033 MD049 MD055 -->
+<!--
+Note to contributors: All guidelines have an anchor tag to allow cross-referencing from associated tooling.
+The anchor tags within a section using a common prefix to ensure uniqueness with anchor tags in other sections.
+Please ensure that you add an anchor tag to any new guidelines that you add and maintain the naming convention.
+-->
+
+## History
+
+<details>
+  <summary>Expand change history</summary>
+
+| Date        | Notes                                                          |
+| ----------- | -------------------------------------------------------------- |
+| 2024-Sep-17 | Added guidance on Secrets                                      |
+
+</details>
+
+## Introduction
+
+These are prescriptive guidelines that Radius contributors MUST follow while designing APIs to ensure that customers have a great experience. These guidelines help make Radius APIs:
+- Developer friendly via consistent patterns
+- Efficient & cost-effective
+
+Technology and software is constantly changing and evolving, and as such, this is intended to be a living document. [Open an issue](https://github.com/radius-project/design-notes/issues) to suggest a change or propose a new idea. 
+
+### Prescriptive Guidance
```

Comment:
I'm wondering if simplifying it down to three tiers of guidance might be more effective? i.e. DO, MAYBE, DO NOT

Reply:
discussed: keeping existing layers. remove language on exceptions from DO and DO NOT